tests/tenancy_test.go

```diff
@@ -0,0 +1,150 @@
+package tests
+
+import (
+	"fmt"
+	"github.com/mesosphere/kudo-spark-operator/tests/utils"
+	log "github.com/sirupsen/logrus"
+	"gotest.tools/assert"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+	"testing"
+)
+
+func TestTenancyTwoOperatorsDifferentNamespaces(t *testing.T) {
+	operators := operatorBuilder(2, true, true)
+	for _, operator := range operators {
+		err := operator.InstallSparkOperator(true)
+		assert.NilError(t, err)
+	}
+
+	verifyComponents(t, operators)
+	verifyWorkloads(t, operators)
+
+	// verify CRDs are present after one of the operators is deleted
+	operators[0].CleanUp()
+	assert.Assert(t, crdsInstalled(t), "CRDs are not present!")
+
+	// check that CRDs are deleted after no operator instances left
+	operators[1].CleanUp()
+	assert.Assert(t, !crdsInstalled(t), "CRDs are not deleted!")
+}
+
+func TestTenancyTwoOperatorsSingleNamespace(t *testing.T) {
+	operators := operatorBuilder(2, false, true)
+	for _, operator := range operators {
+		err := operator.InstallSparkOperator(false)
+		assert.NilError(t, err)
+		defer operator.CleanUp()
+	}
+
+	verifyComponents(t, operators)
+	verifyWorkloads(t, operators)
+}
+
+func TestTenancyTwoOperatorsSameNameDifferentNamespaces(t *testing.T) {
+	operators := operatorBuilder(2, true, false)
+	for _, operator := range operators {
+		err := operator.InstallSparkOperator(true)
+		assert.NilError(t, err)
+	}
+
+	verifyComponents(t, operators)
+	verifyWorkloads(t, operators)
```

Comment:
Here and in the above method:
```suggestion
t.Run("TestWorkloads", func(t *testing.T) {
	verifyWorkloads(t, operators)
})
```